player/tests.py

```diff
@@ -0,0 +1,50 @@
+from django.contrib.auth import get_user_model
+from .models import Player, BallGame
+import pytest
+
+
+@pytest.mark.django_db
+class TestPlayerModel:
+
+    @pytest.fixture
+    def user(self):
+        return get_user_model().objects.create_user(
+            username='testuser',
+            password='testpass'
+        )
+
+    @pytest.fixture
+    def saved_player(self, user):
+        player = Player.objects.create(
+            user=user,
+            birth_date='1990-01-01',
+            favorite_ball_game=BallGame.Soccer
+        )
+        return player
+
+    def test_get_player(self, saved_player):
+        player = Player.objects.get(pk=saved_player.pk)
+        assert player == saved_player
+
+    def test_create_player(self, saved_player, user):
+        player = saved_player
+        assert player.user == user
+        assert player.birth_date == '1990-01-01'
+        assert player.favorite_ball_game == BallGame.Soccer
+
+    def test_update_player(self, saved_player):
+        player = saved_player
+        player.favorite_ball_game = BallGame.Basketball
+        player.save()
+        updated_player = Player.objects.get(pk=player.pk)
+        assert updated_player.favorite_ball_game == BallGame.Basketball
+
+    def test_delete_player(self, saved_player):
+        player = saved_player
+        player.delete()
+        assert not Player.objects.filter(pk=player.pk).exists()
+
+    def test_delete_user_deletes_player(self, saved_player):
+        user = saved_player.user
+        user.delete()
+        assert not Player.objects.filter(pk=saved_player.pk).exists()
```

Comment:
I'd make this assertion as closest as possible for being the exact opposite of the one verifying that the player was created. This way, you're asserting in the same way that it existed and that it doesn't exist once you remove it.

I think you should get an exception if you try to fetch it by its pk, so just the fetching inside the relevant context manager. Something like:

```suggestion
        with pytest.raises(TheTypeOfTheExpectedException):
                Player.objects.get(pk=saved_player.pk)
```

Reply:
Making the two checks different always raise the concern whether you actually changed something. i.e. you need to be convinced that this line did not return the same value before you called delete.